src/app/header/header.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, Output, EventEmitter, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.css']
+})
+export class HeaderComponent {
+
+  openCabinet: boolean = false
+
+  @Output()
+  changeAuthHidding = new EventEmitter()
+
+  constructor() { }
+
+  @Input()
+  loggedIn!: boolean;
+
+  // Hide "Sign in" btn and chenge "Add new ad" btn 
+  changeAuth(): void {
+    this.changeAuthHidding.emit(true)
+  }
+
+  // Open menu (as cabinet)
+  openCabinetFunc(value: boolean): void {
```

Comment:
this function is redundant. You may use it in template. Please review all cases that might exist as well